lib/widgets/mxc_image.dart

```diff
@@ -72,7 +79,9 @@ class _MxcImageState extends State<MxcImage>
     final cacheKey = widget.cacheKey;
     cacheKey == null
         ? _imageDataNoCache = data
-        : _imageDataCache[cacheKey] = data;
+        : widget.cacheMap != null
+            ? widget.cacheMap![cacheKey] = data
+            : _imageDataCache[cacheKey] = data;
```

Comment:
a bit tough to understand, could you use some "if" please ?  😊

Reply:
updated